### PR TITLE
Fedora

### LINKED ISSUE #38: [mirscreencast] segfaults after "Permission denied" error on Fedora
Seen on both Fedora 26 (real, but old x386 hardware) and Fedora 27 VM. (It might "just" be the drivers.)

Message: Failed to mmap buffer
Location : mesa/client/client_buffer.cpp(64)

### PR DESCRIPTION
**Pull request by AlanGriffiths**
Get mirscreencast working on Fedora. (Fixes #38)

### CODE REVIEW

**Comment by Saviq** [comment]
bors r=RAOF

**Comment by AlanGriffiths** [comment]
bors r=RAOF

**Comment by AlanGriffiths** [comment]
bors r=RAOF

**Comment by AlanGriffiths** [comment]
Hmm, the crash didn't magically go away...

[ RUN ] CompositingScreencastTest.captures_to_buffer_by_compositing
/spread/mir/tools/detect_fd_leaks.bash: line 84: 14309 Segmentation fault      $@ 2>&1

Something for Monday morning

**Comment by AlanGriffiths** [comment]
Hmm, mirscreencast needs fixing for the "Mir EGL" path on Ubuntu. (Because creating BufferStreamScreencast no longer fails, the error is only detected when we capture the first buffer.)

File: examples/miral-shell/miral-screencast.sh [hunk 1]
```diff
@@ -4,6 +4,8 @@ height=1080
 output=screencast.mp4
 socket=${XDG_RUNTIME_DIR}/miral_socket
 if [ -v MIR_SERVER ]; then socket=${MIR_SERVER}; fi
+bindir=$(dirname $0)
+if [ "${bindir}" != "" ]; then bindir="${bindir}/"; fi
```
Comment (Saviq) [inline examples/miral-shell/miral-screencast.sh]: Why this?

File: examples/miral-shell/miral-screencast.sh [hunk 2]
```diff
@@ -39,7 +41,7 @@ if [ -e ${output} ]; then echo "Output exists, moving to ${output}~"; mv ${outpu
 while [ ! -e "${socket}" ]; do echo "waiting for ${socket}"; sleep 1 ;done
 
 if [ ! -v tempfile ]; then tempfile=$(mktemp); fi
-mirscreencast --size ${width} ${height} -m ${socket} -f ${tempfile}& mirscreencast_pid=$!
+${bindir}mirscreencast --size ${width} ${height} -m ${socket} -f ${tempfile}& mirscreencast_pid=$!
```
Comment (Saviq) [inline examples/miral-shell/miral-screencast.sh]: I'd put the slash after `}` here.

Comment (AlanGriffiths) [reply examples/miral-shell/miral-screencast.sh]: This approach was C&P from miral-app, but when "$(dirname $0)" == "" putting the slash after "}" results in "/mirscreencast" and the script fails.

(I forget which scenario leads to that case, but I do know I had to fix it in miral-app.)

**Pull request merged**